extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,368 @@
+	 * state (eg, unreviewed, rejected, questionable), order (last edited, priority),
+	 * and timestamp.
+	 *
+	 * When getting this batch of images, we first check if the timestamp matches
+	 * a previously reviewd batch of images. If so, we return those images. If not,
+	 * we'll create a new review assigning those images to the timestamp passed in.
+	 *
+	 * @param int $timestamp
+	 * @param int $state
+	 * @param int $order
+	 * @return array
+	 */
+	public function getImageList( int $timestamp, int $state, int $order ) {
+		if ( $this->previouslyViewedImageList( $timestamp ) ) {
+			return $this->refetchImageListByTimestamp( $timestamp );
+		}
+
+		return $this->fetchNewImageList( $timestamp, $state, $order );
+	}
+
+	private function fetchNewImageList( $timestamp, $state, $order ) {
+		$deleteFromQueueList = [];
+		$imageList = [];
+		$iconsWhere = [];
+
+		// Set a hard and soft limit for stopping the loop.  The soft limit changes
+		// only when an image matches, the hard limit changes for every iteration.
+		$hardNum = self::LIMIT_IMAGES_FROM_DB;
+		$softNum = self::LIMIT_IMAGES;
+
+		$reviewStart = wfTimestamp( TS_DB, $timestamp );
+		$imageIter = $this->getLatestImageIter( $order, $state );
+		while ( $softNum && $hardNum && $row = $imageIter() ) {
+			$hardNum--;
+
+			// Grab this image now to eliminate any race conditions with other reviewers
+			if ( !$this->assignImageToUser( $reviewStart, $state, $row ) ) {
+				// If we failed to assign this image to ourselves, try the next
+				continue;
+			}
+
+			$imageInfo = $this->getImageData( $row );
+
+			// If we failed to load this image, remove it from the review queue
+			if ( !empty( $imageInfo['error'] ) ) {
+				$deleteFromQueueList[] = [
+					'wiki_id' => $row->wiki_id,
+					'page_id' => $row->page_id,
+					'reason' => $imageInfo['error'],
+				];
+				continue;
+			}
+
+			// If this is an icon, put it in a separate queue
+			if ( strpos( 'ico', $imageInfo['extension'] ) ) {
+				$iconsWhere[] = "(wiki_id = {$row->wiki_id} and page_id = {$row->page_id})";
+				continue;
+			}
+
+			$imageList[] = $imageInfo;
+			$softNum--;
+		}
+
+		// Remove invalid images
+		$this->createDeleteFromQueueTask( $deleteFromQueueList );
+
+		// Move icons to different state
+		$this->moveImagesToIcoStateFromWhere( $iconsWhere );
+
+		// Return valid images list
+		WikiaLogger::instance()->info( 'ImageReviewLog', [
+			'method' => __METHOD__,
+			'message' => 'Fetched ' . count( $imageList ) . ' new images',
+		] );
+
+		return $imageList;
+	}
+
+	/**
+	 * @param $timestamp
+	 * @return bool
+	 * @throws Exception
+	 * @throws MWException
+	 */
+	private function previouslyViewedImageList( $timestamp ) {
+		$db = $this->getDatawareDB();
+		return ( new WikiaSQL() )
+			->SELECT( '1' )
+			->FROM( 'image_review' )
+			->WHERE( 'reviewer_id' )->EQUAL_TO( $this->userId )
+			->AND_( 'review_start' )->EQUAL_TO( wfTimestamp( TS_DB, $timestamp ) )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "review_start" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)